### ISSUE TITLE
g-link doesn't load VueRemark content when markdown contains Vue components

### ISSUE THREAD

**Issue by mattbrailsford** [gridsome/gridsome#897]
## Description

I'm creating a docs site for a product I'm working on and when I run the site locally using `gridsome develop` everything works fine. However, when I deploy the site (both locally / on netlify) using `gridsome build` and then test the site I'm seeing an issue where content from `g-link` links isn't loading if the page you start from has a vue component in it's markdown (using VueRemark).

You can see this happen on my staging site. If you start from this page https://xenodochial-ritchie-e79759.netlify.com/docs/payment-providers/stripe/versions/ and then click on the "Changelog" link in the sidebar, you'll see that the main content area doesn't update. Likewise, if you start on the changelog page http://localhost:8953/docs/core/changelog/ and navigate to the "Versions" page, the content won't update. Both of these pages contain markdown that contains vue components. If you look in dev tools, you can see the content actually loads, it just isn't rendered.

Now, if you start on a page that has content not containing vue components, so just plain old markdown such as http://localhost:8953/docs/core/1-0-0/ this loads fine, and then you can navigate to the versions page via the version number next to the product name in the sidebar, and the content for that page now loads fine. As does then navigating to the changelog page.

### Steps to reproduce

1. Create a site with at least two pages
2. In the content of one of the pages, add a globally registered vue component to it
3. Setup links between the two pages using `g-link`
4. Build the site using `gridesome build`
5. Sever the built site and launch in a browser
6. Starting on the page with the vue component, do a browser refresh then click the link to the other page. The address bar should update, but the page content doesn't.
7. Starting on the page without the vue component, do a browser refresh then click the link to the other page. You should now be able to navigate freely between any page on the site.

### Expected result

The page content should update according to the page you are on

### Actual result

If you start your browser session on the page with the vue component in it's markdown, navigating the site won't update the main content area. If you start on a page without a vue component, then everything works fine.

**Comment by mattbrailsford** [comment]
Actually, I've noticed that the page I'm suggesting doesn't contain a vue component, does actually contain one. So I can only deduce this must be because the versions + changelog pages contain vue components that have nested markdown contained within them?

**Comment by mattbrailsford** [comment]
Ok, I've looked further and I can't narrow it down any further, but it seems to be down to having multiple vue components in the markdown, but them not being wrapped by HTML.

If I render multiple vue components, but wrap them in markup, they seem to work. If you go to this page https://xenodochial-ritchie-e79759.netlify.com/docs/payment-providers/stripe/versions/ which is the same as the original versions page, but now I've wrapped the vue components I use with div elements, all of a sudden this page works fine, linking to / from other pages. By the original versions link that had components not wrapped in markup still fails.

**Comment by mattbrailsford** [comment]
Right, I managed to work out what it was. The components I'm using render tables but if I render them without a tbody tag (which I was), I get the errors. If I render them with tbody tags however everything works.

I'm guessing there is a difference between what gets rendered server side and what gets generated for hydration with one generating with tbody tags and the other not.

**Comment by oliverpool** [comment]
I just got bitten by this as well. Example on https://5e1f900a359fdc0008709cfd--dpsg-toelz-83646.netlify.com/ :

- Clicking on the menu items does nothing when you come from the home
- But everything works fine if you start from another page (https://5e1f900a359fdc0008709cfd--dpsg-toelz-83646.netlify.com/gruppenstunden/ for instance)

**My main issue is I can't see any log or warning informing about something wrong (in the markup).**

**Comment by brandonpittman** [comment]
Having a similar issue myself. Not sure what’s causing it for me, but I am using Vue Remark as well.
